src/lib/AST/ASTVisitor.cpp

```diff
@@ -2101,6 +2205,110 @@ class ASTVisitor
 
     //------------------------------------------------
 
+    void
+    buildNamespaceAlias(
+        AliasInfo& I,
+        bool created,
+        NamespaceAliasDecl* D)
+    {
+        bool documented = parseRawComment(I.javadoc, D);
+        addSourceLocation(I, D->getBeginLoc(), true, documented);
+
+        if(! created)
+            return;
+
+        I.Name = extractName(D);
+        if (D->getQualifier())
+        {
+            I.Qualifier = buildNameInfo(D->getQualifier());
+        }
+
+        // A NamedDecl nominated by a NamespaceAliasDecl
+        // will be one of the following:
+        // - NamespaceDecl
+        if(NamedDecl* ND = D->getAliasedNamespace())
+        {
+            SymbolID id;
+            getDependencyID(ND, id);
```

Comment:
You don't need to do all these explicit `SymbolID::invalid` checks, just do `getDependencyID(ND, I.AliasedSymbol);